dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java

```diff
@@ -120,10 +121,15 @@ public static SourceCodeRepoInterface createSourceCodeRepo(String gitUrl, String
      */
     public static Map<String, String> parseGitUrl(String url) {
         // format 1 git@github.com:dockstore/dockstore-ui.git
-        Pattern p1 = Pattern.compile("git\\@(\\S+):(\\S+)/(\\S+)\\.git");
+        // Avoid SonarCloud warning: Using slow regular expressions is security-sensitive
+        // https://sonarcloud.io/organizations/dockstore/rules?open=java%3AS5852&rule_key=java%3AS5852
+        // See Prevent Catastrophic Backtracking and Possessive Quantifiers and Atomic Grouping to The Rescue
+        // in https://www.regular-expressions.info/catastrophic.html
+        // So use more restrictive regex and possesive quantifiers '++' with atomic group '?>'
+        Pattern p1 = Pattern.compile("git@([^\\s:]++):([^\\s/]++)/(?>(\\S+)\\.git$)");
```

Comment:
It was like this, but can you make this and the other pattern static class members. I'm surprised SonarCloud wasn't flagging this already: https://rules.sonarsource.com/java/tag/performance/RSPEC-4248

Reply:
done